cache/disk/disk.go

```diff
@@ -529,37 +550,49 @@ func (c *Cache) Get(kind cache.EntryKind, hash string) (io.ReadCloser, int64, er
 //
 // If there is a local cache miss, the proxy backend (if there is
 // one) will be checked.
-func (c *Cache) Contains(kind cache.EntryKind, hash string) (bool, int64) {
+//
+// The 'size' of the potential content, shall be provided when known,
+// or as -1 when unknown.
+func (c *Cache) Contains(kind cache.EntryKind, hash string, size int64) (bool, int64) {
 
 	// The hash format is checked properly in the http/grpc code.
 	// Just perform a simple/fast check here, to catch bad tests.
 	if len(hash) != sha256HashStrSize {
 		return false, int64(-1)
 	}
 
-	var foundLocally bool
-	size := int64(-1)
+	if kind == cache.CAS && size <= 0 && hash == emptySha256 {
+		return true, 0
+	}
+
+	var found bool
+	foundSize := int64(-1)
 	key := cacheKey(kind, hash)
 
 	c.mu.Lock()
-	val, found := c.lru.Get(key)
+	val, isInLru := c.lru.Get(key)
 	// Uncommitted (i.e. uploading items) should be reported as not ok
-	if found {
+	if isInLru {
 		item := val.(*lruItem)
-		foundLocally = item.committed
-		size = item.size
+		found = item.committed
+		foundSize = item.size
 	}
 	c.mu.Unlock()
 
-	if foundLocally {
-		return true, size
```

Comment:
The reason was to avoid duplicating the size check in both the local and proxy branches. But perhaps your suggestion is more readable and I updated according to that.